crates/gpui/src/platform/windows/dispatcher.rs

```diff
@@ -0,0 +1,106 @@
+use std::thread::{current, JoinHandle, ThreadId};
+
+use async_task::Runnable;
+use flume::Sender;
+use parking::Parker;
+use parking_lot::Mutex;
+
+use crate::{PlatformDispatcher, TaskLabel};
+
+pub(crate) struct WindowsDispatcher {
+    background_sender: Sender<(Runnable, Option<TaskLabel>)>,
+    main_sender: Sender<Runnable>,
+    background_threads: Vec<JoinHandle<()>>,
+    parker: Mutex<Parker>,
+    main_thread_id: ThreadId,
+}
+
+impl WindowsDispatcher {
+    pub(crate) fn new(main_sender: Sender<Runnable>) -> Self {
+        let parker = Mutex::new(Parker::new());
+        let (background_sender, background_receiver) =
+            flume::unbounded::<(Runnable, Option<TaskLabel>)>();
+        let background_threads = (0..std::thread::available_parallelism()
```

Comment:
nice, this seems to create a thread pool. Though unsure if we want an upper limit on number of threads.